### PR TITLE
sqlite dbhash sqldiff sqlite-analyzer 3.41.1

### PR DESCRIPTION
**Pull request by ZhongRuoyu**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by ZhongRuoyu** [comment]
Strange enough, I don't think a `libgda` revbump is needed because there's no linkage error.

```
$ brew linkage libgda
System libraries:
  /lib/x86_64-linux-gnu/libc.so.6
Homebrew libraries:
  /home/linuxbrew/.linuxbrew/opt/glib/lib/libgio-2.0.so.0 (glib)
  /home/linuxbrew/.linuxbrew/opt/glib/lib/libglib-2.0.so.0 (glib)
  /home/linuxbrew/.linuxbrew/opt/glib/lib/libgmodule-2.0.so.0 (glib)
  /home/linuxbrew/.linuxbrew/opt/glib/lib/libgobject-2.0.so.0 (glib)
  /home/linuxbrew/.linuxbrew/Cellar/libgda/6.0.0_1/lib/libgda-6.0.so.6.0.0 (libgda)
  /home/linuxbrew/.linuxbrew/opt/libxml2/lib/libxml2.so.2 (libxml2)
  /home/linuxbrew/.linuxbrew/opt/sqlite/lib/libsqlite3.so.0 (sqlite)
Dependencies with no linkage:
  iso-codes
```

And `libgda`'s test actually passed on Linux with the following output (notice the first line of output):

```
$ ./example
example[14670]: Error opening directory “/home/linuxbrew/.linuxbrew/Cellar/sqlite/3.41.0/lib”: No such file or directory
ref   | name  | price
------+-------+---------
p1    | chair | 2.000000
p2    | table | 5.000000
p3    | glass | 1.100000
p1000 | ???   |     NULL
p1001 | ???   |     NULL
(5 rows)
ref   | name    | price
------+---------+---------
p1    | chair   | 2.000000
p2    | table   | 5.000000
p3    | glass   | 1.100000
p1000 | flowers | 1.990000
p1001 | ???     |     NULL
(5 rows)
ref   | name    | price
------+---------+---------
p1    | chair   | 2.000000
p3    | glass   | 1.100000
p1000 | flowers | 1.990000
(3 rows)
```

The cellar path must have been hardcoded somewhere in `libgda`.

**Comment by ZhongRuoyu** [comment]
Yep:

```
$ strings `brew --prefix libgda`/lib/libgda-6.0.so | grep Cellar
/home/linuxbrew/.linuxbrew/Cellar/sqlite/3.41.0/lib
/home/linuxbrew/.linuxbrew/Cellar/libgda/6.0.0_1/etc
/home/linuxbrew/.linuxbrew/Cellar/libgda/6.0.0_1/libexec
/home/linuxbrew/.linuxbrew/Cellar/libgda/6.0.0_1/lib
/home/linuxbrew/.linuxbrew/Cellar/libgda/6.0.0_1/share
/home/linuxbrew/.linuxbrew/Cellar/libgda/6.0.0_1/sbin
/home/linuxbrew/.linuxbrew/Cellar/libgda/6.0.0_1/bin
/home/linuxbrew/.linuxbrew/Cellar/libgda/6.0.0_1
/home/linuxbrew/.linuxbrew/Cellar/libgda/6.0.0_1/lib:/home/linuxbrew/.linuxbrew/opt/gcc/lib/gcc/current:/home/linuxbrew/.linuxbrew/opt/xz/lib:/home/linuxbrew/.linuxbrew/opt/bzip2/lib:/home/linuxbrew/.linuxbrew/opt/zlib/lib:/home/linuxbrew/.linuxbrew/opt/pcre2/lib:/home/linuxbrew/.linuxbrew/opt/libffi/lib:/home/linuxbrew/.linuxbrew/opt/mpdecimal/lib:/home/linuxbrew/.linuxbrew/opt/openssl@1.1/lib:/home/linuxbrew/.linuxbrew/opt/ncurses/lib:/home/linuxbrew/.linuxbrew/opt/readline/lib:/home/linuxbrew/.linuxbrew/opt/sqlite/lib:/home/linuxbrew/.linuxbrew/opt/expat/lib:/home/linuxbrew/.linuxbrew/opt/libedit/lib:/home/linuxbrew/.linuxbrew/opt/libxcrypt/lib:/home/linuxbrew/.linuxbrew/opt/berkeley-db@5/lib:/home/linuxbrew/.linuxbrew/opt/krb5/lib:/home/linuxbrew/.linuxbrew/opt/libtirpc/lib:/home/linuxbrew/.linuxbrew/opt/libnsl/lib:/home/linuxbrew/.linuxbrew/opt/python@3.11/lib:/home/linuxbrew/.linuxbrew/opt/util-linux/lib:/home/linuxbrew/.linuxbrew/opt/glib/lib:/home/linuxbrew/.linuxbrew/opt/json-glib/lib:/home/linuxbrew/.linuxbrew/opt/icu4c/lib:/home/linuxbrew/.linuxbrew/opt/libxml2/lib:/home/linuxbrew/.linuxbrew/lib
```

**Comment by ZhongRuoyu** [comment]
```
$ cat `brew --prefix sqlite`/lib/pkgconfig/sqlite3.pc
# Package Information for pkg-config

prefix=/home/linuxbrew/.linuxbrew/Cellar/sqlite/3.41.1
exec_prefix=${prefix}
libdir=${exec_prefix}/lib
includedir=${prefix}/include

Name: SQLite
Description: SQL database engine
Version: 3.41.1
Libs: -L${libdir} -lsqlite3
Libs.private: -lz -lm
Cflags: -I${includedir}
```

Not good. The cellar path should be changed to `opt_prefix`.

File: Formula/sqlite.rb [hunk 1]
```diff
@@ -50,6 +50,9 @@ def install
 
     system "./configure", *args
     system "make", "install"
+
+    # Avoid rebuilds of dependants that hardcode this path.
+    inreplace lib/"pkgconfig/sqlite3.pc", prefix, opt_prefix
```
Comment (ZhongRuoyu) [inline Formula/sqlite.rb:55]: With this we should be able to get rid of `libgda` revbumps for future `sqlite` upgrades.

**Review by p-linnane — APPROVED** [review]

**Pull request closed without merging**